translator/source_code/translator_shared_library.py

```diff
@@ -0,0 +1,171 @@
+import xmltodict
+import uuid
+import sys
+import getopt
+from jsonschema import validate
+from jsonschema import ValidationError
+import logging
+from collections import OrderedDict
+import re
+import os
+from datetime import datetime
+import os.path
+import random
+import string
+
+
+def validate_info(info):
+
+    if ((not info) or (type(info) != dict and type(info) != OrderedDict)):
+        logging.warning('invalid type')
+        return False
+
+    feed_info_id = str(info.get('feed_info_id', ''))
+    check_feed_info_id = re.match(
+        '[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}', feed_info_id)
+
+    metadata = info.get('metadata', {})
+    wz_location_method = metadata.get('wz_location_method')
+    lrs_type = metadata.get('lrs_type')
+    contact_name = metadata.get('contact_name')
+    contact_email = metadata.get('contact_email')
+    issuing_organization = metadata.get('issuing_organization')
+    required_fields = [check_feed_info_id, metadata, wz_location_method,
+                       lrs_type, contact_name, contact_email, issuing_organization]
+    for field in required_fields:
+        if not field:
+            logging.warning('Not all required fields are present')
+            return False
+
+    return True
+
+
+def parse_xml(inputfile):
+    if not inputfile or not os.path.isfile(inputfile):
+        return None
+    with open(inputfile, encoding='utf-8-sig') as f:
+        xml_string = f.read()
+        inputfile_obj = xmltodict.parse(xml_string)
+        return inputfile_obj
+
+
+def validate_wzdx(wzdx_obj, wzdx_schema):
+    if not wzdx_schema or not wzdx_obj:
+        return False
+    try:
+        validate(instance=wzdx_obj, schema=wzdx_schema)
+    except ValidationError as e:
+        logging.error(RuntimeError(str(e)))
+        return False
+    return True
+
+
+def initialize_info():
+    info = {}
+    info['feed_info_id'] = "104d7746-688c-44ed-b195-2ee948bf9dfa"
```

Comment:
We're still hard coding this feed info id to the same value for both cotrip and icone data. This would then make our output data appear to be coming from the same location, which doesn't seem correct.

Reply:
should be fixed